### ISSUE TITLE
Crash during plan/apply

### ISSUE THREAD

**Issue by mackoftrack** [hashicorp/terraform#22496]
### Terraform Version

```
Terraform v0.12.6
+ provider.google v2.13.0
+ provider.google-beta v2.13.0
+ provider.mongodbatlas (unversioned)
+ provider.null v2.1.2
+ provider.template v2.1.2

```

### Terraform Configuration Files

Not sure what's causing this crash so i'll just post the logs
```hcl

```

### Crash Output

```
Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Unavailable desc = transport is closing

panic: runtime error: invalid memory address or nil pointer dereference
2019-08-20T09:38:58.824-0700 [DEBUG] plugin.terraform-provider-google-beta_v2.13.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x21 pc=0x19a9727]
2019-08-20T09:38:58.824-0700 [DEBUG] plugin.terraform-provider-google-beta_v2.13.0_x4:
2019-08-20T09:38:58.824-0700 [DEBUG] plugin.terraform-provider-google-beta_v2.13.0_x4: goroutine 90 [running]:
```

Full crash.log here:
https://gist.github.com/75c008f0b7da9630c7707258e7c82b87

**Comment by mackoftrack** [comment]
Can anyone help? I'm kind of dead-in-the-water here...

**Comment by umegaya** [comment]
@mackoftrack did you use `google_sql_database_instance` resource and specify `""` for `settings.backup_configuration.start_time`? 

recent days (within a week), this setting causes crash at https://github.com/terraform-providers/terraform-provider-google/blob/master/vendor/github.com/hashicorp/terraform/terraform/diff.go#L522 because diff become `<nil>`, I suspect that some minor change of google API response causes buggy behaviour of terraform diff logic. 

for my case, setting proper value ("HH:MM" format) to start_time stops this crash. hope this helps 👍

**Comment by jbardin** [comment]
Closed by #22690
